src/installer/installer.go

```diff
@@ -71,41 +70,52 @@ func (i *installer) InstallNode() error {
 		return err
 	}
 
-	i.UpdateHostInstallProgress(models.HostStageStartingInstallation, i.Config.Role)
-
-	i.log.Infof("Start cleaning up device %s", i.Device)
-	err := i.cleanupInstallDevice()
-	if err != nil {
-		i.log.Errorf("failed to prepare install device %s, err %s", i.Device, err)
+	isBootstrap := i.Config.Role == string(models.HostRoleBootstrap)
+	isSingleNodeBootstrap := isBootstrap && i.SingleNode
+	if err := i.startInstallation(isSingleNodeBootstrap); err != nil {
 		return err
 	}
-	i.log.Infof("Finished cleaning up device %s", i.Device)
 
-	if err = i.ops.Mkdir(InstallDir); err != nil {
-		i.log.Errorf("Failed to create install dir: %s", err)
-		return err
-	}
 	ctx, cancel := context.WithCancel(context.Background())
 	errs, _ := errgroup.WithContext(ctx)
 	//cancel the context in case this method ends
 	defer cancel()
-	isBootstrap := false
-	if i.Config.Role == string(models.HostRoleBootstrap) {
-		isBootstrap = true
+
+	if isBootstrap {
 		errs.Go(func() error {
 			return i.runBootstrap(ctx)
 		})
 		i.Config.Role = string(models.HostRoleMaster)
 	}
 
+	if err := i.installNode(isSingleNodeBootstrap); err != nil {
```

Comment:
Just don't call installNode if `isSingleNodeBootstrap == true`
```suggestion
	if isSingleNodeBootstrap {
            i.log.Infof("Bootstrap in single node mode, skipping installNode")
	} else {
	    if err := i.installNode(); err != nil {
```